pandas/core/indexes/accessors.py

```diff
@@ -219,6 +219,40 @@ def to_pydatetime(self) -> np.ndarray:
     def freq(self):
         return self._get_values().inferred_freq
 
+    @property
+    def isocalendar(self):
+        """
+        Returns a DataFrame with the year, week, and day calculated according to
+        the ISO 8601 standard.
+
+        .. versionadded:: 1.1.0
+
+        Returns
+        -------
+        DataFrame
+            with columns year, week and day
+
+        See Also
+        --------
+        Timestamp.isocalendar
+        datetime.date.isocalendar
+
+        Examples
+        --------
+        >>> pd.to_datetime(pd.Series(["2020-01-01"])).dt.isocalendar
```

Comment:
I'd just create one datetime `Series` in the beginning and just reference that in the other examples:

```
>>> ser = pd.Series(pd.to_datetime(["2020-01-01", "NaT"]))
>>> ser
...

>>>  ser.dt.isocalendar
>>>  ser.dt.isocalendar.week
```